### ISSUE TITLE
[Accessibility]: Insufficient contrast in search result expansion button

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#43145]
### Audit type

ARC Toolkit

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34190

### Problem description

The "Show $x more matches" button has insufficient contrast in both dark and light mode

#### Dark mode
<img width="307" alt="CleanShot 2022-10-18 at 15 55 28@2x" src="https://user-images.githubusercontent.com/206864/196560301-8f456870-9dad-47ac-b445-352214ee5098.png">

![image](https://user-images.githubusercontent.com/206864/197251495-f822e954-7bd7-48f4-b36d-36e9322ca3ca.png)

#### Light mode
![image](https://user-images.githubusercontent.com/206864/197251412-52758df3-b85b-43ea-b060-1d1311d33ea6.png)

![image](https://user-images.githubusercontent.com/206864/197251461-33e1f3c7-846f-40eb-8e11-78942f34e241.png)

### Expected behavior

@almeidapaulooliveira to suggest an appropriate color, then assign to @limitedmage to fix.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

/cc @lguychard

**Comment by almeidapaulooliveira** [comment]
@limitedmage 

### Dark theme:
- New text color: Root/$gray-05 (Figma), --gray-05 (CSS)

### Light theme:
- New text color: Root/$gray-08 (Figma), --gray-08 (CSS)

Let me know if it works.

**Comment by limitedmage** [comment]
These colors are perfect, thanks @almeidapaulooliveira!

**Comment by quinnkeast** [comment]
@almeidapaulooliveira can you assign this new colour pair to a new semantic colour in Wildcard?

/cc @AlicjaSuska

**Comment by limitedmage** [comment]
@quinnkeast @almeidapaulooliveira @AlicjaSuska  I've named this color `--collapse-results-color` in the PR. Happy to change it to something else.

**Comment by almeidapaulooliveira** [comment]
Thanks, @limitedmage. I created #43741 to update the Wildcard with the changes introduced with the 4.0 and 4.1 releases.

**Comment by AlicjaSuska** [comment]
@limitedmage --collapse-results-color is very specific. Is it used for anything else in the system? It seems like it's a color for backgrounds with less contrast to our regular text colors. Maybe something like --text-high-contrast. @quinnkeast ?

**Comment by limitedmage** [comment]
@AlicjaSuska I don't think this combination of colors for dark/light mode are being used currently in our system anywhere else.
